### ISSUE TITLE
[BUG] Filename broken when use reverse proxy

### ISSUE THREAD

**Issue by NoraH1to** [perry-mitchell/webdav-client#342]
To solve the web app cross-domain problem, I use vercel for reverse proxy with the following rules:

```json
{
  "rewrites": [
    // reverse proxy
    { "source": "/proxy/:match/:url*", "destination": "https://:match/:url*" },
    {
      "source": "/((?!proxy\/.*).*)",
      "destination": "/index.html"
    }
  ]
}
```

I use this proxy to create a webDAV client like:

```typescript
const client = createClient('https://my-proxy.vercel.app/proxy/webdav.server.com/dav');
```

Suppose the root directory of the webdav service has the following contents:

```
file.txt
```

Then I request the contents of the root directory `client.getDirectoryContents('/')`.

The `filename` of `file.txt` should be `"/dav/file.txt"`.

But it actually provides `"/../../../dav/file.txt"`.

I think the problem is here: https://github.com/perry-mitchell/webdav-client/blob/master/source/operations/directoryContents.ts#L75-L78

Is there any way to fix it?

**Comment by NoraH1to** [comment]
Due to cross-domain issues, many serverless web apps that use third-party services require reverse proxy, which can cause the problem mentioned above.

I think it can be solved by adding `directoryBasePath` field to the client options, which allow user to customize the real service base path without affecting other existing features.

I have forked the repository and implemented the feature ready for commit if need.

**Comment by perry-mitchell** [comment]
Hi! Sounds like an interesting case and definitely something I'd like working correctly.

I'd be interested in seeing an MR to add such an option - would you be interested in making one?

Also, would you imagine that this could be replicated in an integration test so that we can ensure it works as expected and solves a potentially common problem?

**Comment by NoraH1to** [comment]
> I'd be interested in seeing an MR to add such an option - would you be interested in making one?

Of course, I've already submitted #343.

> Also, would you imagine that this could be replicated in an integration test so that we can ensure it works as expected and solves a potentially common problem?

The test is also in there, we just need to make sure we use it to resolve the file path when the user pass it.

**Comment by perry-mitchell** [comment]
Released under 5.1.0
